operator/duties/attester.go

```diff
@@ -0,0 +1,265 @@
+package duties
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	eth2apiv1 "github.com/attestantio/go-eth2-client/api/v1"
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	spectypes "github.com/bloxapp/ssv-spec/types"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/bloxapp/ssv/logging/fields"
+)
+
+type AttesterHandler struct {
+	baseHandler
+
+	duties            *Duties[phase0.Epoch, *eth2apiv1.AttesterDuty]
+	fetchCurrentEpoch bool
+	fetchNextEpoch    bool
+}
+
+func NewAttesterHandler() *AttesterHandler {
+	return &AttesterHandler{
+		duties: NewDuties[phase0.Epoch, *eth2apiv1.AttesterDuty](),
+	}
+}
+
+func (h *AttesterHandler) Name() string {
+	return spectypes.BNRoleAttester.String()
+}
+
+// HandleDuties manages the duty lifecycle, handling different cases:
+//
+// On First Run:
+//  1. Fetch duties for the current epoch.
+//  2. If necessary, fetch duties for the next epoch.
+//  3. Execute duties.
+//
+// On Re-org:
+//
+//	If the previous dependent root changed:
+//	    1. Fetch duties for the current epoch.
+//	    2. Execute duties.
+//	If the current dependent root changed:
+//	    1. Execute duties.
+//	    2. If necessary, fetch duties for the next epoch.
+//
+// On Indices Change:
+//  1. Execute duties.
+//  2. Reset duties for the current epoch.
+//  3. Fetch duties for the current epoch.
+//  4. If necessary, fetch duties for the next epoch.
+//
+// On Ticker event:
+//  1. Execute duties.
+//  2. If necessary, fetch duties for the next epoch.
+func (h *AttesterHandler) HandleDuties(ctx context.Context, logger *zap.Logger) {
```

Comment:
Let's pass logger to NewAttesterHandler instead of passing it to each of its methods